opm/parser/eclipse/EclipseState/SimulationConfig/tests/ThresholdPressureTest.cpp

```diff
@@ -181,13 +181,10 @@ static DeckPtr createDeck(const ParseContext& parseContext , const std::string&
     return parser.parseString(input , parseContext);
 }
 
-static Eclipse3DProperties getEclipseProperties(const Deck& deck)
-{
-    static const EclipseGrid eclipseGrid( 3, 3, 3 );
-    return Eclipse3DProperties( deck, TableManager( deck ), eclipseGrid );
+static Eclipse3DProperties getEclipseProperties(const Deck& deck) {
+    return Eclipse3DProperties(deck, *new TableManager(deck), *new EclipseGrid(3, 3, 3));
```

Comment:
I stil think it is not good practice to leak (anywhere), and it would not clutter up the test, but make them more useful (in the sense of telling the user how to use the class). I made a quick mockup here:
```
struct Setup
{
    ParseContext parseContext;
    DeckPtr deck;
    TableManager tablemanager;
    EclipseGrid grid;
    Eclipse3DProperties props;
    SimulationConfig simulationConfig;

    explicit Setup(const std::string& input)
        : parseContext(),
          deck(createDeck(parseContext, input)),
          tablemanager(*deck),
          grid(10, 3, 4),
          props(*deck, tablemanager, grid),
          simulationConfig(parseContext, *deck, props)
    {
    }
};
```
With this Setup class (it is close to what is usually called a fixture, but not quite since it will be different from case to case) the tests are a little simpler, too:
```
BOOST_AUTO_TEST_CASE(SimulationConfigGetThresholdPressureTableTest) {
    Setup s(inputStr);
    SimulationConfigConstPtr simulationConfigPtr;
    BOOST_CHECK_NO_THROW(simulationConfigPtr = std::make_shared
            <const SimulationConfig>(s.parseContext, *s.deck, s.props));
}


BOOST_AUTO_TEST_CASE(SimulationConfigNOTHPRES) {
    Setup s(inputStr_noTHPRES);
    BOOST_CHECK( !s.simulationConfig.hasThresholdPressure() );
}
```

I do not think this is a messier than leaking.